bundles/org.openhab.binding.boschindego/src/main/java/org/openhab/binding/boschindego/internal/handler/BoschAccountHandler.java

```diff
@@ -91,13 +100,54 @@ public void initialize() {
 
     @Override
     public void dispose() {
-        oAuthFactory.ungetOAuthService(this.getThing().getUID().getAsString());
+        oAuthFactory.ungetOAuthService(thing.getUID().getAsString());
```

Comment:
I still have a doubt that you can later simply use `oAuthFactory.getOAuthClientService` in initialize to get a ready oAuth service. `oAuthFactory.ungetOAuthService` closes the service and releases stuff.
Please check to be sure.

Reply:
```
    public @Nullable OAuthClientService getOAuthClientService(String handle) {
        OAuthClientService clientImpl = oauthClientServiceCache.get(handle);

        if (clientImpl == null || clientImpl.isClosed()) {
            // This happens after reboot, or client was closed without factory knowing; create a new client
            // the store has the handle/config data
            clientImpl = OAuthClientServiceImpl.getInstance(handle, oAuthStoreHandler, tokenExpiresInBuffer,
                    httpClientFactory);
            if (clientImpl == null) {
                return null;
            }
            oauthClientServiceCache.put(handle, clientImpl);
        }
        return clientImpl;
    }
```
Looks like you are right, `getOAuthClientService` will create a new instance in case it was previously closed (in `dispose`).